playwright-e2e/dsm/pages/miscellaneous-pages/user-and-permissions-page.ts

```diff
@@ -0,0 +1,307 @@
+
+    for (let index = 0; index < amountOfPermissions; index++) {
+      const permissionCheckbox = allPossiblePermissions[index];
+      const permission = permissionCheckbox.locator(`//span[@class='ng-star-inserted']`);
+      const currentPermission = (await permission.innerText()).trim();
+      //console.log(`Current permission: ${currentPermission}`);
+      const currentPermissionCastedAsEnum = currentPermission as UserPermission;
+      const verifiedPermission = Object.values(this.currentStudyPermissions).indexOf(currentPermissionCastedAsEnum) >= 0;
+      if (!verifiedPermission) {
+        throw new Error(`The ${currentPermission} permission is not a valid permission for ${study}`);
+      }
+    }
+  }
+
+  public async assertSelectedPermissions(studyAdmin: Locator, permissions: UserPermission[]): Promise<void> {
+    const amountOfPermissionstoCheckFor = permissions.length;
+
+    for (let index = 0; index < amountOfPermissionstoCheckFor; index++) {
+      const checkbox = this.getPermissionCheckbox(studyAdmin, permissions[index]);
+      await expect(checkbox).toBeChecked();
+    }
+  }
+
+  private getPermissionCheckbox(studyAdmin: Locator, permission: UserPermission, forSelection?: boolean): Locator {
+    const permissionName = permission as string;
+    console.log(`Permission Name: ${permissionName}`);
+    if (forSelection) {
+      //below xpath works best for selection/clicking
+      return studyAdmin.locator(`//ancestor::mat-expansion-panel//app-permission-checkbox[contains(.,'${permissionName}')]//div`)
+    }
+    //below xpath works best for determining if the checkbox is currently selected
+    return studyAdmin.locator(`//ancestor::mat-expansion-panel//app-permission-checkbox[contains(.,'${permissionName}')]//input`);
+  }
+
+  /* Utility methods */
+  public async getStudyAdminName(name: string): Promise<string> {
+    const studyAdminName = this.page.getByText(`${name}`);
+    return studyAdminName.innerText();
+  }
+
+  public async getStudyAdminPhoneNumber(phoneNumber: string): Promise<string> {
+    const studyAdminPhoneNumber = this.page.getByText(`${phoneNumber}`);
+    return studyAdminPhoneNumber.innerText();
+  }
+
+  private isCMIStudy(study: StudyEnum): boolean {
+    return study === (
+      StudyEnum.ANGIO ||
+      StudyEnum.BRAIN ||
+      StudyEnum.ESC ||
+      StudyEnum.MBC ||
+      StudyEnum.OSTEO ||
+      StudyEnum.PANCAN ||
+      StudyEnum.PROSTATE)
+  }
+
+  private isPECGSStudy(study: StudyEnum): boolean {
+    return study === (
+      StudyEnum.OSTEO2 ||
+      StudyEnum.LMS
+    )
+  }
+
+  private determineStudyGroup(study: StudyEnum): string {
+    let studyGroup = '';
+
+    if (this.isCMIStudy(study)) {
+      //Study is a CMI research study
+      studyGroup = this.CMI_STUDY_GROUP;
+    } else if (this.isPECGSStudy(study)) {
+      //Study is a CMI clinical study
+      studyGroup = this.PECGS_STUDY_GROUP;
+    } else if (study === StudyEnum.RGP) {
+      //Study is the RGP study
```

Comment:
I think these comments are redundant.  The code is nice and clear as is without comments :)